mlflow/tracking/_model_registry/utils.py

```diff
@@ -150,6 +152,19 @@ def _get_databricks_rest_store(store_uri, **_):
     return RestStore(partial(get_databricks_host_creds, store_uri))
 
 
+def _get_databricks_uc_rest_store(store_uri, **_):
+    from mlflow.exceptions import MlflowException
+    from mlflow.version import VERSION
+
+    raise MlflowException(
+        f"Detected Unity Catalog model registry URI '{mlflow.get_registry_uri()}'. "
+        f"However, the current version of the MLflow client ({VERSION}) does not support models "
+        f"in the Unity Catalog. Please upgrade to the latest version of the MLflow Python client "
```

Comment:
In which MLflow version are we going to support the UC registry store?

Reply:
Maybe I'm misunderstanding something.

Reply:
Good question - it actually depends on whether we can land the remaining PRs before the next OSS MLflow branch cut. In case we can't (given the branch cut is coming soon & we still haven't completed the backend implementation or gotten to test the client e2e), I wanted to land this PR because otherwise `mlflow.set_registry_uri("databricks-uc")` silently logs to a file at the relative path `./databricks-uc` on the local filesystem, which would be confusing to users